src/ui/control/geolocate_control.js

```diff
@@ -53,6 +53,9 @@ function checkGeolocationSupport(callback) {
     }
 }
 
+let numberOfWatches = 0;
+let noTimeout = false;
```

Comment:
What confused me when reviewing is I thought these would be scoped for each control, but it seems they are shared globally across the page (which is why they work). I think it's worth adding a comment to explain what these are for and what they represent, and mention something about tracking number of watches on the page, even when multiple maps and geolocate controls may be present.

Reply:
yeah they have to be scoped to the global control state so that we can track when multiple watches are being created. i will update with a comment